parser/src/lexer.rs

```diff
@@ -513,6 +536,18 @@ where
                             Some('t') => {
                                 string_content.push('\t');
                             }
+                            Some('u') => match self.uncicode_literal(4) {
+                                Ok(c) => string_content.push(c),
+                                Err(e) => return Err(e),
+                            },
+                            Some('U') => match self.uncicode_literal(8) {
+                                Ok(c) => string_content.push(c),
+                                Err(e) => return Err(e),
+                            },
+                            Some('x') if !is_bytes => match self.uncicode_literal(2) {
+                                Ok(c) => string_content.push(c),
+                                Err(e) => return Err(e),
```

Comment:
The pattern `Err(e) => return Err(e)` can be short circuit like this:
```rust
Some('x') if !is_bytes => string_content.push(self.unicode_literal(2)?),
```

Reply:
Thanks.  I refactor this with `?`.